src/devtools/client/debugger/src/components/Editor/ColumnBreakpoints.tsx

```diff
@@ -26,33 +29,59 @@ type PropsFromRedux = ConnectedProps<typeof connector>;
 
 type $FixTypeLater = any;
 interface CBProps {
-  editor: $FixTypeLater;
+  editor: SourceEditor;
 }
 
-class ColumnBreakpoints extends Component<PropsFromRedux & CBProps> {
-  render() {
-    const { cx, editor, columnBreakpoints, selectedSource } = this.props;
-
-    if (!selectedSource || columnBreakpoints.length === 0) {
-      return null;
-    }
-
-    let breakpoints;
-    // TODO Is this a safe thing to do while rendering?
-    editor.codeMirror.operation(() => {
-      breakpoints = columnBreakpoints.map((breakpoint, i) => (
-        <ColumnBreakpoint
-          cx={cx}
-          key={getLocationKey(breakpoint.location)}
-          columnBreakpoint={breakpoint}
-          editor={editor}
-          source={selectedSource}
-          insertAt={i}
-        />
-      ));
-    });
-    return <div>{breakpoints}</div>;
+function ColumnBreakpoints2({ editor }: CBProps) {
+  const cx = useAppSelector(getContext);
+  const selectedSource = useAppSelector(getSelectedSource);
+  const columnBreakpoints = useAppSelector(getVisibleColumnBreakpoints);
+  const { value: hitCountsMode } = useStringPref("hitCounts");
+
+  useEffect(() => {
```

Comment:
Would `useLayoutEffect` fit?

Reply:
_Could_ certainly use that here, but I'm not sure there's any real benefit. Almost all of the CSS var updates will occur later as the CodeMirror editor is resized.  The only difference that would make here is right after the initial render, and after the "Show Hit Counts" option is changed.

Reply:
Agreed with Mark.

Since we're calling `updateWidth` to check for any updates we may have missed between render and mount (something we should be doing even with layout effect) then this is fine. There's a potential for a small visible flicker if the timing is just right (wrong?)